src/v/kafka/client/client.h

```diff
@@ -112,7 +113,15 @@ class client {
     ss::future<produce_response>
     produce_records(model::topic topic, chunked_vector<record_essence> batch);
 
-    ss::future<list_offsets_response> list_offsets(model::topic_partition tp);
+    ss::future<list_offsets_response> list_offsets(
+      chunked_vector<model::topic_partition> tps,
```

Comment:
Do you think it might be worth having:
```c++
chunked_hash_map<model::topic, chunked_vector<model::partition_id>>
// Or
chunked_hash_map<model::topic, chunked_hash_set<model::partition_id>>
```
Mostly to avoid the duplicated topic names? Although it would also avoid some linear searches.

Alternatively we just expose `list_offsets_request`.